helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +300,213 @@ fn request_completion(
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<rope::Regex> =
+        Lazy::new(|| rope::Regex::new(r"((?:~|\$HOME|\$\{HOME\})?(?:\.{0,2}/)+.*)").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let start = text.line_to_char(cur_line).max(cursor.saturating_sub(1000));
+    let line_until_cursor = text.slice(..).regex_input_at(start..cursor);
+
+    let (dir_path, typed_file_name) = PATH_REGEX
+        .find_iter(line_until_cursor)
+        .last() // this is a workaround for missing "match end of slice" support in `rope::Regex`
+        .and_then(|matched_path| {
+            let end = text.byte_to_char(matched_path.end());
+            if end != cursor {
+                return None;
+            }
+            let start = text.byte_to_char(matched_path.start());
+            let matched_path = &text.slice(start..end).to_string();
+
+            // resolve home dir (~/, $HOME/, ${HOME}/) on unix
+            #[cfg(unix)]
+            let mut path = {
+                static HOME_DIR: Lazy<Option<std::ffi::OsString>> =
+                    Lazy::new(|| std::env::var_os("HOME"));
+
+                let home_resolved_path = if let Some(home) = &*HOME_DIR {
+                    let first_separator_after_home = if matched_path.starts_with("~/") {
+                        Some(1)
+                    } else if matched_path.starts_with("$HOME") {
+                        Some(5)
+                    } else if matched_path.starts_with("${HOME}") {
+                        Some(7)
+                    } else {
+                        None
+                    };
+                    if let Some(start_char) = first_separator_after_home {
+                        let mut path = home.to_owned();
+                        path.push(&matched_path[start_char..]);
+                        path
+                    } else {
+                        matched_path.into()
+                    }
+                } else {
+                    matched_path.into()
+                };
+                PathBuf::from(home_resolved_path)
+            };
+            #[cfg(not(unix))]
+            let mut path = PathBuf::from(matched_path);
+
+            if path.is_relative() {
+                if let Some(doc_path) = doc.path().and_then(|dp| dp.parent()) {
+                    path = doc_path.join(path);
+                } else if let Ok(work_dir) = std::env::current_dir() {
+                    path = work_dir.join(path);
+                }
+            }
+            let ends_with_slash = match matched_path.chars().last() {
+                Some(std::path::MAIN_SEPARATOR) => true,
+                None => return None,
+                _ => false,
+            };
+            // check if there are chars after the last slash, and if these chars represent a directory
+            match std::fs::metadata(path.clone()).ok() {
```

Comment:
It could well be that this is an artifact, after a [recent change](https://github.com/helix-editor/helix/pull/2608#discussion_r1740152557), I've overlooked, I'll take a look into it.

Reply:
I removed that, is the current version better?